src/main/java/Duke.java

```diff
@@ -1,10 +1,378 @@
+        //         text = text.substring(6, text.length());
+        //         int indexOfSlash = text.indexOf("/",9);
+        //         String day = text.substring(indexOfSlash+4, text.length());
+        //         String task = text.substring(0, indexOfSlash);
+        //         text = task + "(at: " + day + ")";
+
+        //         System.out.println("  [E][ ] " + text);
+        //         System.out.println("Now you have " + count + " tasks in the list.");
+        //         System.out.println(line);
+        //         tasks[count] = text;
+        //         type[count] = "[E]";
+        //         marks[count] = "[ ]";
+        //         count++;
+        //     }
+        // }
+
+        // Level 5
+
+        String line = "---------------------------------------------";
+        String greet = "Hello! I'm Duke";
+        String question = "What can I do for you?";
+        System.out.println(line + "\n" + greet + "\n" + question + "\n" + line);
+
+        boolean run = true;
+        String[] tasks = new String[100];
+        String[] marks = new String[100];
+        String[] type = new String[100];
+        int count = 1;
+
+        while(run){
+            Scanner in = new Scanner(System.in);
+            String text;
+            text = in.nextLine();
+            if(text.equals("bye")){
+                System.out.println(line);
+                System.out.println("Bye. Hope to see you again soon!");
+                System.out.println(line);
+                run = false;
+            }
+            else if (text.equals("list")){
+                System.out.println(line);
+                System.out.println("Here are the tasks in your list:");
+                for(int i = 1; i< count; i++){
+                    System.out.println(i + ". " + type[i] + marks[i] + " " + tasks[i]);
+                }
+                System.out.println(line);
+            }
+
+            else if (text.contains("mark")){
+                System.out.println(line);
+                String strNum;
+                int num;
+                if(text.contains("unmark")){ //Unmark
+                    System.out.println("OK, I've marked this task as not done yet:");
+                    strNum = text.substring(7, text.length());
+                    num = Integer.parseInt(strNum);
+                    marks[num] = "[ ]";
+                    System.out.println(marks[num] + " " + tasks[num]);
+
+                }
+                else{ //Mark
+                    System.out.println("Nice! I've marked this task as done:");
+                    strNum = text.substring(5, text.length());
+                    num = Integer.parseInt(strNum);
+                    marks[num] = "[X]";
+                    System.out.println(marks[num] + " " + tasks[num]);
+                }
+
+                System.out.println(line);
+            }
+
+            else if (text.contains("todo")) {
+                if(text.equals("todo") || text.equals("todo ")){
```

Comment:
Avoid complicated expressions. Consider breaking up the logic into different lines of code.